uvtools/dspec.py

```diff
@@ -1,78 +1,160 @@
 import aipy
-import binning
 import numpy as np
 
 def wedge_width(bl_len, sdf, nchan, standoff=0., horizon=1.):
-    '''Return the (upper,lower) delay bins that geometrically
-    correspond to the sky, for bl_len in ns, sdf in GHz, and the number
-    of channels nchan.'''
+    '''Return the (upper,lower) delay bins that geometrically correspond to the sky.
+    Variable names preserved for backward compatability with capo/PAPER analysis.
+    
+    Arguments:
+        bl_len: length of baseline (in 1/[sdf], typically ns)
+        sdf: frequency channel width (typically in GHz)
+        nchan: number of frequency channels
+        standoff: fixed additional delay beyond the horizon (same units as bl_len)
+        horizon: proportionality constant for bl_len where 1 is the horizon (full light travel time)
+
+    Returns:
+        uthresh, lthresh: bin indices for upper and lower edges of the fourier filter
+    '''
     bl_dly = horizon * bl_len + standoff
-    bin_dly = 1. / (sdf * nchan)
-    w = int(round(bl_dly/bin_dly))
+    return calc_width(bl_dly, sdf, nchan)    
+
+
+def calc_width(filter_size, real_delta, nsamples):
+    '''Calculate the upper and lower bin indices of a fourier filter.
+
+    Arguments:
+        filter_size: the half-width (i.e. the width of the positive part) of the region in fourier 
+            space, symmetric about 0, that is filtered out. In units of 1/[real_delta].
+        real_delta: the bin width in real space
```

Comment:
Does this need units?

Reply:
No. It's general. The filter size has units that depend on it's units though.